### PR TITLE
Upload files for hns rename folder test in gcs bucket 

### PR DESCRIPTION
**Pull request by anushka567**
### Description
Helper function to upload files in the gcs bucket at specified location
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - [link](https://github.com/GoogleCloudPlatform/gcsfuse/blob/1aa474c936244fb8683f4f861e55fc2d457ba145/perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files_test.py#L260)
3. Integration tests - NA

### CODE REVIEW

**Review by ankitaluthra1 — APPROVED** [review]

File: perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py [hunk 1]
```diff
@@ -50,30 +55,30 @@ def _check_for_config_file_inconsistency(config) -> (int):
...
   if "name" not in config:
-    _logmessage("Bucket name not specified")
+    _logmessage("Bucket name not specified","error")
     return 1
 
   if "folders" in config:
     if not ("num_folders" in config["folders"] or "folder_structure" in config[
       "folders"]):
-      _logmessage("Key missing for folder")
+      _logmessage("Key missing for nested folder","error")
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py:64]: Create const for error

File: perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py [hunk 2]
```diff
@@ -238,16 +243,63 @@ def _delete_existing_data_in_gcs_bucket(gcs_bucket)->(int):
...
+        if (file_size_unit.lower() == 'mb'):
+          out.truncate(1024 * 1024 * int(file_size))
+        if (file_size_unit.lower() == 'kb'):
+          out.truncate(1024 * int(file_size))
+        if (file_size_unit.lower() == 'b'):
+          out.truncate(int(file_size))
+
+    num_files = os.listdir(TEMPORARY_DIRECTORY)
+
+    if not num_files:
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py:274]: Compare num_files with actual number

File: perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py [hunk 3]
```diff
@@ -238,16 +243,63 @@ def _delete_existing_data_in_gcs_bucket(gcs_bucket)->(int):
...
+
+    # Starting upload to the gcs bucket.
+    try:
+        subprocess.Popen(
+        'gcloud storage cp --recursive {}/* {}'.format(TEMPORARY_DIRECTORY,
+                                         destination_blob_name),
+        shell=True).communicate()
+    except subprocess.CalledProcessError as e:
+      _logmessage("Issue while uploading files to GCS bucket.Aborting...","error")
+      return -1
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py:286]: exit 1

File: perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files_test.py [hunk 4]
```diff
@@ -305,5 +306,121 @@ def test_deleting_success(self,mock_check_output):
...
+    temp_file='./tmp/data_gen/file_1.txt'
+
+    exit_code = generate_folders_and_files._generate_files_and_upload_to_gcs_bucket(
+        destination_blob_name, num_of_files, file_size_unit, file_size,
+        filename_prefix)
+
+    # Assert that temp_file is opened.
+    mock_open.assert_called_once_with(temp_file, 'wb')
+    # Assert that 'truncate' was called with the expected size.
+    expected_size = 1024 * 1024 * int(file_size)
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files_test.py:341]: Initialise it above.

File: perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py (hunk 2 above)
Comment (anushka567) [reply perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py:274]: Will get complicated when number of files to be created is not a multiple of 100.

**Review by Tulsishah — APPROVED** [review]

**Review by ankitaluthra1 — APPROVED** [review]
lgtm

**Pull request merged**